File: IsometricShooter/Scripts/CS/Material/MaterialAnimation.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class MaterialAnimation : MonoBehaviour {

	public MeshRenderer materialTarget;
	public List<Material> materialList; //list of Material
	public float animationRate = 1.0f;
	public bool activateAnimation = true;
	public bool stopIfRigidBodyZeroVelocity = true;
	public Rigidbody rigidBodyTarget;
	
	private bool isAnimating;
	private Material originalMaterial;
	private int frame = 0;
	private float nextFrame = 0.0f; //helper for frame rate

	// Use this for initialization
	void Start () {
	
		originalMaterial = materialTarget.material;
	
	}
	
	// Update is called once per frame
	void Update () {
	
		if(activateAnimation) {
		
			if(stopIfRigidBodyZeroVelocity) {
				if(rigidBodyTarget.velocity.magnitude <= 0.1f) { //small range to avoid animation at "nearly" zero velocity
					isAnimating = false;
				} else {
					isAnimating = true;
				}
			}
			
			if(isAnimating) {
				if(Time.time > nextFrame) {
					nextFrame = Time.time + animationRate;
					materialTarget.material = materialList[frame];
					if(frame < materialList.Count-1) {
						frame++;
					} else {
						frame = 0;
					}
				}
			} else {
				materialTarget.material = originalMaterial;
			}
			
		}
	
	}
}

File: IsometricShooter/Scripts/JS/Material/MaterialAnimation.js
#pragma strict

import System.Collections.Generic; //enable lists

var materialTarget : MeshRenderer;
var materialList : List.<Material>; //list of Material
var animationRate : float = 1.0;
var activateAnimation : boolean = true;
var stopIfRigidBodyZeroVelocity : boolean = true;
var rigidBodyTarget : Rigidbody;

private var isAnimating : boolean;
private var originalMaterial : Material;
private var frame : int = 0;
private var nextFrame : float = 0.0; //helper for frame rate

function Start () {

	originalMaterial = materialTarget.material;

}

function Update () {

	if(activateAnimation) {
	
		if(stopIfRigidBodyZeroVelocity) {
			if(rigidBodyTarget.velocity.magnitude <= 0.1) { //small range to avoid animation at "nearly" zero velocity
				isAnimating = false;
			} else {
				isAnimating = true;
			}
		}
		
		if(isAnimating) {
			if(Time.time > nextFrame) {
				nextFrame = Time.time + animationRate;
				materialTarget.material = materialList[frame];
				if(frame < materialList.Count-1) {
					frame++;
				} else {
					frame = 0;
				}
			}
		} else {
			materialTarget.material = originalMaterial;
		}
		
	}

}